src/GitVersion.Configuration.Tests/Configuration/ConfigurationProviderTests.CanWriteOutEffectiveConfiguration.approved.txt

```diff
@@ -11,6 +11,8 @@ commit-date-format: yyyy-MM-dd
 merge-message-formats: {}
 update-build-number: true
 semantic-version-format: Strict
+strategies:
+- ConfigNext, MergeMessage, TaggedCommit, TrackReleaseBranches, VersionInBranchName
```

Comment:
```suggestion
strategies: [ConfigNext, MergeMessage, TaggedCommit, TrackReleaseBranches, VersionInBranchName]
```

Reply:
Actually the deserialize gives me this format. I have this not in control

Reply:
I know. Which is why we want to separate our domain classes from DTO objects used for de/serialization. We can just have a private class that we use for serialization and deserialization, close to where it is used in the perimeter of the codebase, and then map from that class to the core domain object. In our mapping code, we will do the translation between array and `[Flags] enum`.

Reply:
I think you are mistaken here. This is equivalent in yaml and the property is of type VersionStrategies[] already:

```yaml
strategies:
- ConfigNext, MergeMessage, TaggedCommit, TrackReleaseBranches, VersionInBranchName

strategies:
- ConfigNext
- MergeMessage
- TaggedCommit
- TrackReleaseBranches
- VersionInBranchName
```

Reply:
Hm, I've never seen arrays like this before:

```yaml
strategies:
- ConfigNext, MergeMessage, TaggedCommit, TrackReleaseBranches, VersionInBranchName
```

I don't see it [in this documentation of YAML arrays](https://www.w3schools.io/file/yaml-arrays/) either. And if you try to [this online converter](https://onlineyamltools.com/convert-yaml-to-json), it turns this:

<img width="575" alt="yaml" src="https://github.com/GitTools/GitVersion/assets/12283/31ec3eed-ae12-4e1e-948b-da673b44abc5">

into this:

<img width="588" alt="json" src="https://github.com/GitTools/GitVersion/assets/12283/9873638d-0f61-4cb2-b4fc-eb5c86598128">

As you can see from the converted JSON:

```json
{
  "strategies": [
    "ConfigNext, MergeMessage, TaggedCommit, TrackReleaseBranches, VersionInBranchName"
  ]
}
```

`strategies` here has a single item which is the comma separated string `ConfigNext, MergeMessage, TaggedCommit, TrackReleaseBranches, VersionInBranchName`. That's not what we want, is it?

Reply:
Okay I think I need to take a look. That is indeed very strange

Reply:
At the moment it looks like the following:

![image](https://github.com/GitTools/GitVersion/assets/56404113/a54d05f7-654d-4d82-a76d-f9d3e62498fc)

Reply:
Yep, that looks good.